Library/Homebrew/cmd/formulae.sh

```diff
@@ -8,5 +8,17 @@
 source "${HOMEBREW_LIBRARY}/Homebrew/items.sh"
 
 homebrew-formulae() {
-  homebrew-items '*\.rb' 'Casks' 's|/Formula/|/|' '^homebrew/core'
+  local formulae
+  formulae="$(homebrew-items '*\.rb' 'Casks' 's|/Formula/|/|' '^homebrew/core')"
+
+  # HOMEBREW_CACHE is set by brew.sh
+  # shellcheck disable=SC2154
+  if [[ -n "${HOMEBREW_INSTALL_FROM_API}" && -f "${HOMEBREW_CACHE}/api/formula.json" ]]
+  then
+    local api_formulae
+    api_formulae="$(ruby -e "require 'json'; JSON.parse(File.read('${HOMEBREW_CACHE}/api/formula.json')).each { |f| puts f['name'] }" 2>/dev/null)"
```

Comment:
I wonder if we should do this calculation when we download `HOMEBREW_CACHE/api/formula.json`, instead. That way we only need to do it once per unique cache file (I'm assuming this line is relatively slow)

Reply:
@Rylan12 That seems like a good idea! Let's merge as-is for now and iterate to improve performance here 💪🏻